cw-orch/src/interchain/follow_ibc_execution.rs

```diff
@@ -0,0 +1,211 @@
+///     In the logs, we can find all details that allow us to identify the transaction in which the packet is received in the distant chain
+///     These include :
+///     - The connection_id
+///     - The destination port
+///     - The destination channel
+///     - The packet sequence (to identify a specific packet in the channel)
+///
+///     ## Remarks
+///     - The packet data is also retrieved for logging
+///     - Multiple packets can be sent out during the same transaction.
+///         In order to identify them, we assume the order of the events is the same for all events of a single packet.
+///         Ex: packet_connection = ["connection_id_of_packet_1", "connection_id_of_packet_2"]
+///         Ex: packet_dst_port = ["packet_dst_port_of_packet_1", "packet_dst_port_of_packet_2"]
+///     - The chain id of the destination chain is not available directly in the logs.
+///         However, it is possible to query the node for the chain id of the counterparty chain linked by a connection
+///
+/// 2. Receive packet. For each packet received, we use the identification of the packet that was sent in the original tx to find the tx in which the packet was received
+///     We make sure that only one transaction tracks receiving this packet.
+///         If not, we sent out an error (this error actually comes from the code not identifying an IBC packet properly)
+///         If such an error happens, it means this function is not implemented properly
+///         We verify this transaction is not errored (it should never error)
+///     
+/// 3. Acknowledgment. The last part of the packet lifetime is the acknowledgement the distant chain sents back.
+///     a. Identify acknowledgement
+///         In the same transaction as the one in which the packet is received, an packet acknowledgement should be sent back to the origin chain
+///         We get this acknowledgment and deserialize it according to https://github.com/cosmos/cosmos-sdk/blob/v0.42.4/proto/ibc/core/channel/v1/channel.proto#L134-L147
+///         If the acknowledgement doesn't follow the standard, we don't mind and continue
+///     b. Identify the acknowledgement receive packet on the origin chain
+///         Finally, we get the transaction hash of the transaction in which the acknowledgement is received on the origin chain.
+///         This is also logged for debugging purposes
+///
+/// 4. Finally, some additionnal packets may have been sent out during the whole process. We need to check the generated transactions for additional packets.
+///    This 4th step make the whole process recursive. Those two transactions potentially sends out additional packets !
+///         - The receive Packet transaction identified in 2. on the distant chain
+///         - The receive Acknowledgement transaction identified in 3.b. on the origin chain
+
+#[async_recursion::async_recursion]
+pub async fn follow_trail(chain1: String, channel1: Channel, tx_hash: String) -> Result<()> {
```

Comment:
This could use a different name, "await_ibc_execution"? 

Also tx_hash is what tx specifically? Maybe rename to "packet_send_tx_hash"?

also in regards to IBC the term channel is easy to confuse with gRPC channel, so maybe a rename to grpc_channel1 would be nice as well.

Reply:
tx_hash is the transaction you want to inspect for outgoing packets once it has been transmitted